python-avd/pyavd/_eos_designs/structured_config/network_services/port_channel_interfaces.py

```diff
@@ -42,7 +45,119 @@ def port_channel_interfaces(self: AvdStructuredConfigNetworkServicesProtocol) ->
+
+                    interface_name = l3_port_channel.name
+                    is_subinterface = "." in interface_name
+
+                    if not is_subinterface:
+                        # This is a regular Port-Channel (not sub-interface)
+                        regular_l3_port_channel_names.add(interface_name)
+
+                    parent_port_channel_name = interface_name.split(".", maxsplit=1)[0]
+                    subif_parent_port_channel_names.add(parent_port_channel_name)
+
+                    if is_subinterface:
+                        # Validation for l3_port_channel subinterface
+                        if l3_port_channel.member_interfaces:
+                            msg = f"L3 Port-Channel sub-interface '{interface_name}' has 'member_interfaces' set. This is not a valid setting."
+                            raise AristaAvdInvalidInputsError(msg)
+                        if l3_port_channel._get("mode"):
+                            # implies 'mode' is set when not applicable for a sub-interface
+                            msg = f"L3 Port-Channel sub-interface '{interface_name}' has 'mode' set. This is not a valid setting."
+                            raise AristaAvdInvalidInputsError(msg)
+                        if l3_port_channel._get("mtu"):
+                            # implies 'mtu' is set when not applicable for a sub-interface
+                            msg = f"L3 Port-Channel sub-interface '{interface_name}' has 'mtu' set. This is not a valid setting."
+                            raise AristaAvdInvalidInputsError(msg)
+
+                    # Generate their structured config for the l3_port_channels.
+                    port_channel_interface = EosCliConfigGen.PortChannelInterfacesItem(
+                        name=l3_port_channel.name,
+                        peer=l3_port_channel.peer,
+                        mtu=l3_port_channel.mtu if self.shared_utils.platform_settings.feature_support.per_interface_mtu else None,
+                        shutdown=not l3_port_channel.enabled,
+                        eos_cli=l3_port_channel.raw_eos_cli,
+                        flow_tracker=self.shared_utils.new_get_flow_tracker(
+                            l3_port_channel.flow_tracking, output_type=EosCliConfigGen.EthernetInterfacesItem.FlowTracker
+                        ),
+                        vrf=vrf.name if vrf.name != "default" else None,
+                        peer_type="l3_port_channel",
+                        peer_interface=l3_port_channel.peer_port_channel if l3_port_channel.peer_port_channel else None,
+                    )
+
+                    if l3_port_channel.ipv4_acl_in:
+                        port_channel_interface._update(access_group_in=l3_port_channel.ipv4_acl_in)
+                    if l3_port_channel.ipv4_acl_out:
+                        port_channel_interface._update(access_group_out=l3_port_channel.ipv4_acl_out)
+
+                    if "." not in l3_port_channel.name:
```

Comment:
Let's set a boolean at the top `is_subinterface = "." in l3_port_channel.name` and then use that everywhere.

Reply:
done